src/core/tests/Test_StatusHandlerTruncation.py

```diff
@@ -101,37 +104,44 @@ def test_only_assessment_patches_over_size_limit_truncated(self):
         """ Perform truncation on assessment patches and substatus status is set to Error (not warning) due to per-existing patching errors
         Before truncation: 1000 assessment patches in status, 6 exceptions
         completed status file byte size: 188kb
-        Expected (After truncation): ~669 assessment patches in status
+        Expected (After truncation): ~666 assessment patches in status
+        tombstone records: 2 (Security, Other),
         operation: Assessment,
         assessment substatus name: PatchAssessmentSummary,
         assessment substatus status: error,
         assessment errors code: 1 (error),
         assessment errors details count: 5,
         assessment errors details code: [PACKAGE_LIST_TRUNCATED, OPERATION_FAILED]
-        count of assessment patches removed: 331,
+        count of assessment patches removed: 334,
         truncated status file byte size: 126kb. """
 
         self.__test_scenario = 'assessment_only'
-        self.__patch_count_assessment = 1000
+        patch_count_other = 300
+        patch_count_security = 300
+        patch_count_critical = 400
+        self.__patch_count_assessment = patch_count_other + patch_count_security + patch_count_critical
 
-        self.__set_up_status_file(run='assessment', config_operation=Constants.ASSESSMENT, patch_count=self.__patch_count_assessment, classification='Security')
+        self.__write_assessment_to_status_file(config_operation=Constants.ASSESSMENT, patch_count_assessment=patch_count_critical, patch_count_security=patch_count_security, patch_count_other=patch_count_other)
 
         # Set up complete status file before exceptions
         complete_substatus_file_data = self.__get_substatus_file_json(self.runtime.execution_config.complete_status_file_path)
 
         # Assert complete status file size > 128kb and no exceptions
         self.__assert_patch_summary_from_status(complete_substatus_file_data, Constants.ASSESSMENT, Constants.PATCH_ASSESSMENT_SUMMARY, 'transitioning', self.__patch_count_assessment)
 
-        # Assert status file < 126kb and substatus status remain 'transitioning'
+        # Assert status file < 126kb and substatus status remain 'transitioning', with 2 tombstones (Security, Other)
         truncated_substatus_file_data = self.__get_substatus_file_json(self.runtime.execution_config.status_file_path)
-        self.__assert_patch_summary_from_status(truncated_substatus_file_data, Constants.ASSESSMENT, Constants.PATCH_ASSESSMENT_SUMMARY, 'transitioning', self.__patch_count_assessment, complete_substatus_file_data=complete_substatus_file_data, is_under_internal_size_limit=True, is_truncated=True)
+        self.__assert_patch_summary_from_status(truncated_substatus_file_data, Constants.ASSESSMENT, Constants.PATCH_ASSESSMENT_SUMMARY, 'transitioning', self.__patch_count_assessment + 2, complete_substatus_file_data=complete_substatus_file_data, is_under_internal_size_limit=True, is_truncated=True)
 
```

Comment:
If status file < 126kb, that means, truncation has happened so the total expected patch count should be 666 - 334 

self.__patch_count_assessment is set to 1000 in line 132 
i.e. self.__patch_count_assessment = patch_count_other + patch_count_security + patch_count_critical

How is the expected patch_count 1002 post truncation then, shouldn't it be 332?

Reply:
Haven't gone through the remaining tests beyond this but this could be applicable to other tests too

Reply:
because of the tombstones, input is 1000, but bottom-line total after truncation is 1002 (because we are expecting 2 tmbstone), post truncation w/0 tombstone is 332, but since we have 2 tmbstone, then we need to remove 2 more regular patches, so 334. math should be 666 + 2 + 334 = 1002, hope this make sense?

![image](https://github.com/Azure/LinuxPatchExtension/assets/127874208/d20d1c23-f144-4a2f-b9d0-c6a65ba43e1f)